### PR TITLE
vet_ clinic: Add Multiple tables  and perform queries using join 

### PR DESCRIPTION
**Pull request by Irirwanirira**
## This Pr consists of:

- [ ] Displayed animals belong to Melody Pond?
- [ ] Displayed List of all animals that are pokemon (their type is Pokemon).

- [ ] Listed all owners and their animals, remember to include those that don't own any animal.

- [ ] Displayed animals that are there per species.
- [ ] Lists all Digimon owned by Jennifer Orwell.
- [ ] Listed all animals owned by Dean Winchester that haven't tried to escape.

- [ ] Who owns the most animals?

### Thank you for reviewing my project

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
 
--- Create column species
-ALTER TABLE animals ADD species VARCHAR(40);
+--  species table 
+
+DROP TABLE IF EXISTS species;
+
+CREATE TABLE species (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    names CHAR(100),
```
Comment (Nemwel-Boniface) [inline schema.sql]: - [x] This column should be name and not names

image

File: data.sql [hunk 2]
```diff
@@ -1,34 +1,65 @@
...
+--  insert data for species 
+
+INSERT INTO species(names) VALUES('Pokemon');
+INSERT INTO species(names) VALUES('Digimon');
+
+
+-- Modify your inserted animals so it includes the species_id value
+
+UPDATE animals SET species_id =1 WHERE species_id IS NULL; 
+UPDATE animals SET species_id = 2 WHERE names LIKE '%mon';
```
Comment (Nemwel-Boniface) [inline data.sql]: - [ ] Kindly put this before the previous line. You ideally want to populate the fields, then now add all those fields that are blank.

```
UPDATE animals SET species_id = 2 WHERE names LIKE '%mon';
UPDATE animals SET species_id =1 WHERE species_id IS NULL; 
```

File: queries.sql [hunk 3]
```diff
@@ -79,3 +79,20 @@ SELECT MIN(weight_kg) FROM animals;
 
 -- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
 SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth  BETWEEN '1990-01-01' AND '2000-12-31';
+
+
+-- Write queries (using JOIN) to answer the following questions
+
+SELECT names, full_name FROM animals INNER JOIN owners ON animals.owners_id = owners.id WHERE owners.full_name = 'Melody Pond';
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] Kindly include its screenshot.

image

File: queries.sql [hunk 4]
```diff
@@ -79,3 +79,20 @@ SELECT MIN(weight_kg) FROM animals;
...
+
+SELECT animals.names, species.names FROM animals INNER JOIN species ON animals.species_id = species.id WHERE species.names = 'Pokemon';
+
+SELECT owners.full_name, animals.names FROM owners LEFT JOIN animals ON owners.id = animals.owners_id;
+
+SELECT COUNT(animals.names), species.names FROM animals INNER JOIN species ON animals.species_id = species.id GROUP BY species.names;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] Kindly include its screenshot.

image

File: queries.sql [hunk 5]
```diff
@@ -79,3 +79,20 @@ SELECT MIN(weight_kg) FROM animals;
...
+
+SELECT owners.full_name, animals.names, species.names FROM owners INNER JOIN animals ON owners.id = animals.owners_id INNER JOIN species ON species.id = animals.species_id WHERE species.names = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] This query is wrong. You are supposed to select from animals table and join the species table. Kindly ake this change.

- [x] Kindly include its screenshot.

image

File: queries.sql [hunk 6]
```diff
@@ -79,3 +79,20 @@ SELECT MIN(weight_kg) FROM animals;
...
+
+SELECT owners.full_name, animals.names, animals.escape_attempts FROM owners INNER JOIN animals ON owners.id = animals.owners_id WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [ ] This query is wrong. You are supposed to select from animals table and join the owners table. You can also see so by checking your query results screenshot which is returning a wrong result.  Kindly ake this change.

File: queries.sql [hunk 7]
```diff
@@ -79,3 +79,20 @@ SELECT MIN(weight_kg) FROM animals;
...
+
+SELECT owners.full_name, COUNT(*) FROM owners INNER JOIN animals ON owners.id = animals.owners_id GROUP BY owners.full_name ORDER BY count DESC LIMIT 1;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] This query is wrong. You are supposed to select from animals table and join the owners table. You can also see so by checking your query results screenshot which is returning a wrong result.  Kindly ake this change.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Irirwanirira        :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Irirwanirira) [reply schema.sql]: @Nemwel-Boniface I like your reviews mn, even if they give me a headache.

File: data.sql (hunk 2 above)
Comment (Irirwanirira) [reply data.sql]: @Nemwel-Boniface Thank you for your time.

File: queries.sql (hunk 4 above)
Comment (Irirwanirira) [reply queries.sql]: Done

File: queries.sql (hunk 3 above)
Comment (Irirwanirira) [reply queries.sql]: Thank you 
pond

File: queries.sql (hunk 5 above)
Comment (Irirwanirira) [reply queries.sql]: Thank you for your time.
orwell

File: queries.sql (hunk 6 above)
Comment (Irirwanirira) [reply queries.sql]: @yes thank you;
winchetser

File: queries.sql (hunk 7 above)
Comment (Irirwanirira) [reply queries.sql]: @Nemwel-Boniface  thank you for your time.
most

**Review by mikethreels — APPROVED** [review]
Hi Joseph, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Nemwel-Boniface) [reply schema.sql]: Haha @Irirwanirira  well, :clinking_glasses:  to more reviews then :rofl: 

Good job making all the requested changes. :clap:

**Pull request merged**